osgar/drivers/logzeromq.py

```diff
@@ -13,31 +13,16 @@ class LogZeroMQ:
     def __init__(self, config, bus):
         bus.register('raw:null')
         mode = config['mode']
-        endpoint = config['endpoint']
+        self.endpoint = config['endpoint']
+        self.timeout = config.get('timeout', 1) # default recv timeout 1s
```

Comment:
Well, it is not only moving the config inside the `run()` but also force new timeout (which we already discussed several months ago), so there is no way to set it as blocking and Thread is no longer as `daemon=False`). The behavior is the same as before or how you could have it in multiple threads? Thanks m.

Reply:
Yes, the timeout - not having the timeout set makes it not behaving correctly wrt SIGINT. On the other hand having it does not break anything. Actually, having it set has no manifestation to the outside except properly exiting on SIGINT.

Wrt daemon threads - there is no point in setting any thread to daemon, since `<module>.join()` should wait on all threads the given module has started anyway. So yes, the behavior is the same.

The last part I don't understand:
> how you could have it in multiple threads?

Could you please rephrase it? Thanks.

Reply:
What I meant is that there is always one thread using the socket even in the old implementation ... but it is not the same thread which is creating it. I could imagine problems with slots again (in transmitter) but they are not solved here too?

Reply:
There was `slot_raw` which thankfully has not been used in any config file we have. I considered having the slot available as an invitation to shoot myself in the foot so I removed it as well. It could have been connected to 2 different nodes in which case it would have been called from 2 different threads. It was a disaster waiting to happen :wink: